app/controllers/course/controller.rb

```diff
@@ -16,6 +16,17 @@ def sidebar_items(type: nil)
     sidebar_items_of_type(type).sort_by { |item| weights_hash[item[:key]] || item[:weight] }
   end
 
+  # Fetches the first popup UserNotification for the current course and returns JSON data for the
+  # frontend to display it.
+  def next_popup_notification
+    notification =
+      current_user.notifications.popup.ordered_by_updated_at.unread.includes(activity: :object).
+      drop_while { |popup| popup.activity.object.course_id != current_course.id }.first
+
+    notification && render_to_string("#{helpers.notification_view_path(notification)}.json",
```

Comment:
Is the return value here false or nil? Does it matter if we pass these kind of values into the json render?

Reply:
If notification is `nil`, the `data` field will be omitted. The front end will read it as `null` and the React node will not be rendered.

Have added documentation.